substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/typestate/MultiTypeState.java

```diff
@@ -332,8 +333,12 @@ public void noteMerge(BigBang bb) {
 
     @Override
     public boolean closeToAllInstantiated(BigBang bb) {
-        if (typesCount > 200 && bb != null) {
-            MultiTypeState allInstState = (MultiTypeState) bb.getAllInstantiatedTypeFlow().getState();
+        if (!(bb instanceof PointsToAnalysis)) {
+            return false;
```

Comment:
`MultiTypeState` is a `pointsto` class, can this condition ever be true?

Reply:
I've been thinking about the possibility of reusing the `TypeState` classes even outside the points-to analysis. An abstraction for sets of types and operations on them sounds useful, but these classes currently contain quite a lot of points-to specific code for that. If the time comes, I can try to refactor it, but that is currently a non-goal, so I'll just remove the check for now.